### PR TITLE
Add default value to SearchResults.cursor

### PR DESCRIPTION
**Pull request by sgrimm**
Having to explicitly set the search results cursor to `null` adds clutter to
tests. Make the field default to null and remove the explicit null values from
all the tests. Tests that pass non-null values (to test that cursors work) are
unchanged here.

### CODE REVIEW

**Comment by sgrimm** [comment]
* **#1938** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1938?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @sgrimm and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by sgrimm** [comment]
A couple notes on this:

1. The edit to `SearchResults.kt` is manual. Everything else is automated search-and-replace using IntelliJ's structural search.
2. I know there are PRs pending with new search test cases; reviewing them is what prompted me to do this, though honestly this change was long overdue anyway. I will wait until those PRs are merged before merging this one, and before merging this, I'll rebase it and rerun the structural search and replace to remove the nulls from the new tests. So no need to change the existing PRs to match this.

**Review by tommylau523 — APPROVED** [review]
Looks good to me! I am working on adding deliverables and submissions to search too but I will leave our cursor on those tests.

**Pull request merged**
